language/functional_tests/tests/testsuite/natives/vector/vector_swap.mvir

```diff
@@ -0,0 +1,23 @@
+import 0x0.Vector;
+
+main() {
+  let vec: Vector.T<u64>;
+  let is_empty: bool;
+  vec = Vector.empty<u64>();
+
+  // non-empty case
+  Vector.push_back<u64>(&mut vec, 0);
+  Vector.push_back<u64>(&mut vec, 1);
+  Vector.push_back<u64>(&mut vec, 2);
+  Vector.push_back<u64>(&mut vec, 3);
+
+  Vector.swap<u64>(&mut vec, 0, 3);
+  Vector.swap<u64>(&mut vec, 1, 2);
+  assert(*Vector.borrow<u64>(&vec, 0) == 3, 1);
+  assert(*Vector.borrow<u64>(&vec, 1) == 2, 2);
+  assert(*Vector.borrow<u64>(&vec, 2) == 1, 3);
+  assert(*Vector.borrow<u64>(&vec, 3) == 0, 4);
+
+  return;
+
+}
```

Comment:
Could you add some tests to show that:

- swapping with an out-of-bounds index fails
- swapping with the same index does the expected thing?